src/metabase/actions/execution.clj

```diff
@@ -80,7 +78,7 @@
                          :type                   qp.error-type/invalid-parameter
                          :parameters             request-parameters
                          :destination-parameters (:parameters action)}))))
-    (actions/check-actions-enabled! database)
+    (actions/check-actions-enabled-for-database! action)
```

Comment:
why is it taking an action, but not `database`?

Reply:
It's an optimization. It does the some thing as before but now with only one DB query, and it's all wrapped up in `actions/check-actions-enabled!`.